src/merkle/store/mod.rs

```diff
@@ -257,6 +257,26 @@ impl MerkleStore {
         Ok(tree_depth)
     }
 
+    // DATA EXTRACTORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a subset of this Merkle store such that the returned Merkle store contains all
+    /// nodes which are descendants of the specified roots.
+    ///
+    /// The roots for which no descendants exist in this Merkle store are ignored.
+    pub fn subset<I, R>(&self, roots: I) -> MerkleStore
```

Comment:
As future improvement, this could be a `MerkleStoreRef<'a>` so we just point to the roots of the base `MerkleStore` - unless we intend to mutate this subset and store it elsewhere

However, this suggested approach wouldn't be optimal for I/O, large storage that, by default, will load subsets (from the disk?) into memory, probably as owned, so the user can manipulate the data.

If we don't target such case (i.e. we intend to go in-memory only), then `MerkleStoreRef<'a>` is ideal

Reply:
Do you mean to use `MerkleStoreRef<'a>` as a return type? Or something else?